### DISCUSSION TITLE
Is the project future at risk?

### DISCUSSION THREAD

**Discussion by thekoma** [fluxcd/flux2#4544] [Q&A] [answered]
I see that Weave Works have finished the budget for the project. Many developers that are full time on Flux2 have no option than to get on other trains or continue on a personal investment.

Is it time to look around? Do we have a donation/business support fee to opt to?

Many companies rely on this project to go on so a little perspective is needed here.

**Comment by webwurst** [comment]
Flux reached [all the criteria](https://github.com/cncf/toc/blob/main/process/graduation_criteria.md#graduation-stage) to become a [CNCF Graduated Project](https://landscape.cncf.io/card-mode?selected=flux). So long term support should be fine. But you are right that [most contributions](https://flux.devstats.cncf.io/d/5/companies-table?orgId=1&var-period_name=Last%20quarter&var-metric=contributions) are provided by Weaveworks. Why do you think this will change?

**Reply by wibed** [reply]
does cncf graduation save you any funding?

**Reply by thekoma** [reply]
> Flux reached [all the criteria](https://github.com/cncf/toc/blob/main/process/graduation_criteria.md#graduation-stage) to become a [CNCF Graduated Project](https://landscape.cncf.io/card-mode?selected=flux). So long term support should be fine. But you are right that [most contributions](https://flux.devstats.cncf.io/d/5/companies-table?orgId=1&var-period_name=Last%20quarter&var-metric=contributions) are providec by Weaveworks. Why do you think this will change?

Because WeaveWorks decided to remove founding to many of the contributors.
Being CNCF graduated does not guarantee any funds AFAIK.

Keeping up with issues, new features and in general managing an open source software with thousands of users is extremely time consuming and complex.

If you want an enterprise grade solution that performs as promised you need to have people working on it 5/7 8/24 at the very least.

To make it clear:
https://youtu.be/mweTc7tDO3I?si=WscF7Oy6Z8S4IOEK&t=00m36s

**Reply by kingdonb** [reply]
LOL I'm not sure if you meant to link to Rick and Morty...

We Flux maintainers are still here, I'm not sure it's fair to make demands on volunteer time like this, we are still working on Flux, we are not behind any schedules in terms of roadmap, and I intend to continue to support Flux.

Though I am not funded by Weaveworks anymore, I hope the situation won't persist too long this way and I'll get a new contract to work on Flux. Please be mindful that individual maintainers are all as committed as we ever were, and while we may be thrown for a loop by some developments, it doesn't benefit anyone to declare the sky is falling without evidence.

Thanks for your concern everyone.

Please see:

* https://github.com/fluxcd/flux2/discussions/4335

So we are aware, Flux maintainers who are affected are not completely broadsided by this, and some discussion has already occurred before the most recent Flux meeting that I think you meant to link to.

Here is that meeting:

https://www.youtube.com/watch?v=73-7hhrJPa8

**Reply by kingdonb** [reply]
We met again today and had discussions that were not recorded, it is the regular weekly Flux Dev Meeting; there are some personal concerns and I cannot share everyone's individual status, but since it's not immediately clear from my part of the discussion, yes, Flux maintainers are by-and-large working to solve this issue together, and some things may be hanging in the balance for a time.

There are many concerns to balance as maintainers are from many countries so the international business aspect is very complex. Getting everyone paid in their own home country may be a trick we cannot pull off without the support of a sophisticated backer.

We would always appreciate support from any Flux community members that wish to get involved, and the best way to get involved is to join us at Bug Scrub or at the Dev meetings: https://fluxcd.io/#calendar is always current with the upcoming Flux meetings.

I don't want to diminish the importance of this thread in any way, or sideline anyone who was thinking they may provide support somehow. Flux is the community as much as it is the codebase! We do need a sustainable plan to maintain in the future, and the community's involvement may become a growing part of that as we continue to evolve. Thanks again, and as I re-watched the Rick and Morty clip, I see now this is definitely not a mistake, what you meant to share :)

**Reply by gruberdev** [reply]
@thekoma 

> Because WeaveWorks decided to remove founding to many of the contributors.

Can you share where I can find this information? 

I am trying to find any statement related to this decision through their communication channels and using search engines, to no avail.

**Reply by kingdonb** [reply]
~~There hasn't been a public statement yet as far as I'm aware, and no idea when one will come. Those of us no longer employed at Weaveworks are not in a position to make statements on behalf of the company.~~

There has been a public statement now, [Weaveworks is closing](https://www.linkedin.com/feed/update/urn:li:activity:7160295096825860096/), but frankly this announcement is about Weaveworks not Flux.

**Reply by thekoma** [reply]
> @thekoma 
> 
> > Because WeaveWorks decided to remove founding to many of the contributors.
> 
> Can you share where I can find this information? 
> 
> I am trying to find any statement related to this decision through their communication channels and using search engines, to no avail.

I keep an eye on LinkedIn profiles of the developers engaged on projects I often work with. I won't quote the developer nor the LinkedIn post as I see how that could be an over-shared information on the wave of the moment.

**Comment by stefanprodan** [comment, marked as answer]
> Many developers that are full time on Flux2 have no option than to get on other trains or continue on a personal investment.

Yes this is accurate. Some of us are looking at ways to keep contributing to Flux. Expect a more official response soon.
Update -> ControlPlane announcement https://control-plane.io/posts/controlplane-backs-the-cncf-flux-project-by-employing-maintainers/

**Reply by stefanprodan** [reply]
Some personal news from me, I have joined @controlplaneio and I will continue working on the Flux project in the same capacity as before, full-time core maintainer.

PS. I'm working on an [Enterprise Distribution for Flux](https://github.com/controlplaneio-fluxcd/distribution) that ControlPlane will offer to sponsor OSS Flux going forward.

**Reply by thekoma** [reply]
That's simply awesome! Really happy for you!

**Reply by arbourd** [reply]
Wow that's amazing! Congratulations to both them and you!

**Reply by svengreb** [reply]
I'm so happy for you and the project! 
Creating an awesome project like this with heart and soul, only to have to give it up, would have been so painful. Flux is by far the best GitOps tool and I'm really happy to hear that it has been "rescued".
I wish you all the best for your new job and hope that all other _WeaveWork'es_ find their ways too.

**Comment by monadic** [comment]
Hi all

This is a call for organisations and people who want to work on Flux in any way, to make themselves known here.  I am posting this having been contacted by 100s of people, including a few really promising offers to help.  I would really like to see such offers in the public domain.  Initially here but I hope we can build up to the Flux website and CNCF as things get more serious.

First off please note that people can help Flux in lots of ways!  Coding, docs, community -- ask the maintainers what they need and how to help, there is really no shortage of stuff.  You can also work on the Flux OSS ecosystem here - https://github.com/weaveworks - and elsewhere.

Second - organisations can get involved too.  End user firms are welcome.   Vendors too!  Including Flux in your own OSS and commercial offerings is a great way to make money.   Eg from: support, consulting, or a larger product.  Or you can hire people to work on and support Flux because that fits your company strategy in some way.   

Prior experience with K8s and Flux isn't a requirement, if you are willing to be a member of this community and roll up your sleeves.  It is a welcoming place.

THIS ISSUE is where to start.  Organisations please say who you are, what you do, and how you'd like to be involved.  Be as concrete as possible, respect that others may now know you, and please include a way for people to connect with you and find out more.

--alexis

**Reply by the-ricker** [reply]
Edgecell (https://edgecell.io) is focused on gitops for edge computing. We will be contributing going forward.

**Reply by jjcallis** [reply]
Happy to help  :)

**Reply by brc859844** [reply]
We (https://teracloud.com/) would be prepared to contribute and/or look at supporting customers, given logical alignment with one of our BU's, and probable customer overlap. Our software group consists of a growing number of entities, with offices in US, Australia, Denmark, Sweden, Germany, Greece, Armenia, and New Zealand. Headcount across the group is currently around 320, primarily in engineering, and includes a global L1/L2 support org. Tech covers pretty much everything from legacy/mainframe through to bleeding edge...

**Reply by muhlba91** [reply]
happy to help! 👍🏻

**Reply by monadic** [reply]
@muhlba91 @jjcallis will you be at kubecon next week?

**Reply by muhlba91** [reply]
@monadic yes, i will be.

**Reply by monadic** [reply]
Wonderful!  Will you be able to come to the f2f ecosystem event on Wednesday 11am?  Please feel free to email me!

**Reply by muhlba91** [reply]
sure, that works for me! i‘m looking forward to seeing you there!

**Comment by paul-carlton** [comment]
I would be prepared to contribute as time permits and given alignment with my new employer's priorities, they are a Flux user so have an interest in its success.

**Comment by mestadler** [comment]
Hi All,

As many of you know, we are continuing to support the CNCF Flux project at Control Plane, we have a growing team of the Core Maintainers who are focused on the continuation of the CNCF Flux Project.  We do offer commercial options for companies that need support beyond what the project provides.  As it would be in bad taste to solicit here in the open source, vendor neutral GitHub, I just want to reinforce what all have said, Flux has a great future, and will continue to innovate as Kubernetes evolves. 

Martin (control-plane.io)

**Comment by bboule** [comment]
Hello All,

My name is Bob Boule the VP of Product for OpsMx I am excited to be posting that OpsMx will also be supporting The Flux Project as part of our OpenCD platform.   We believe in the Open CD concept and allowing the customer to choose based on their specific deployment needs, and when it comes to GitOps deployment, the flux project is a pioneer.  We are adding Flux to the list of supported delivery platforms for our Application Security Modules, Deployment Visibility and Intelligence Modules.  We will also be offering commercial support to Flux Users and actively contributing to the Flux project.  OpsMx has been a supporter and contributor to Open Source CD projects such as Spinnaker and Argo specifically.   We are  Looking forward to working with you all as we get started, please feel free to reach out if there are specific questions or if you would like to engage us on where we can jump in and start helping the community.

By the way, if you are passionate about the Flux project  and looking for your next opportunity we would love to talk to you about opportunities here with OpsMx … drop me a line.

I will be publishing a blog soon that announces our move to support this robust and growing OSS Project and will post here once published.

Bob (opsmx.io)

**Reply by makkes** [reply]
How does your support for Flux align with this announcement from only a couple weeks back? https://www.globenewswire.com/news-release/2024/02/08/2826504/0/en/OpsMx-Announces-New-Enterprise-GitOps-Program-for-Weaveworks-Customers.html

**Reply by bboule** [reply]
@makkes thanks for the response and it is a great question.  Initially when we heard the news around Weaveworks we wanted to step up and offer assistance to these customers by giving them OSS options (at them time we were still evaluating our ability to support Flux) however after talking with users starting to use Flux ourselves we started to ask ourselves what would stop us from adding Flux to our Open CD Stack (both supporting our add on tooling and as a provider of commercial support).  Once we were able to get a little more comfortable with that approach the path became clearer for us in terms of our desire to add Flux to our Open CD offering (currently made up of Spinnaker and Argo CD today).

As you can see here from our CEO: 

https://www.opsmx.com/blog/announcing-opsmx-support-for-cncf-flux-an-important-step-for-open-cd/

And the Blog I wrote last week:

https://www.opsmx.com/blog/introducing-opsmx-support-for-flux-powering-seamless-gitops-workflows/

So we clear up any confusion our initial instinct was to figure out how we could help customers using our current stack, and after some additional consideration we came to the conclusion that   support Flux and the Flux community as an Open CD provider was the right path.

Let me apologize for any confusion, hopefully this helps clear this up and our public statements help re-enforce our long term position around Flux.  We are hoping to form a relationship with the community and support this group by being active members!!

**Comment by RandyAbernethy** [comment]
Hey All,

I'm Randy Abernethy a Managing Partner at RX-M. We do and will continue to offer consulting and training for Flux (https://rx-m.com/cloudnative-training-courses/).

--randy at rx-m.com

**Comment by LucaLanziani** [comment]
Hello everyone, @nearform stongly believes in open source and we will do our best to support flux going forward 💪

**Reply by monadic** [reply]
Luca will you be able to join the ecosystem event on Wednesday please?

**Reply by LucaLanziani** [reply]
I don't see why not, I might have missed the details

**Comment by sionsmith** [comment]
Hi Everyone 👋 

[OSO](https://oso.sh/) offers Enterprise 24/7 support services for all things Kubernetes and GitOps including Flux. We have been a long term partner of Weaveworks and advised 50+ large scale companies on adopting Flux and its surrounding components. Our amazing @jjcallis is a regular in the community and spoke at [GitOps days 2022](https://oso.sh/blog/gitops-days-flux-2/)

We will actively work with other partners in the ecosystem to support, advise and contribute to the Weaveworks projects where we can. Please drop me a message if you wish to discuss these in more detail.

Sion (CTO @osodevops)

**Comment by laszlocph** [comment]
Hey everyone, we are still committed to Flux. Our support these days are sticking to the project, and developing a general purpose UI for it: [gimlet-io/capacitor](https://github.com/gimlet-io/capacitor). As time permits we plan to pick up a couple of Flux issues as well.

**Reply by monadic** [reply]
https://kccnceu2024.sched.com/event/1akzm

**Comment by sudermanjr** [comment]
Flux has had a major impact on the Kubernetes ecosystem, and has helped to bring GitOps into mainstream usage. As an all-K8s-all-the-time company, Fairwinds is eager to ensure that our end users have options to meet their business drivers. We're working to evaluate options to add value and continue to support the community.

**Comment by monadic** [comment]
thanks everyone!

**Comment by lachie83** [comment]
Hi, my name is Lachlan Evenson from Azure. We have both products and customers using Flux. We intend to make contributions to Flux in support of the long-term health of the project alongside our other CNCF project investments. With KubeCon EU coming up in a couple of weeks, it might be a good opportunity for us to connect as a community in person (and virtually) to further discuss this topic. I'm lachie83 on CNCF slack.

**Reply by puja108** [reply]
That's a great idea. Happy to have our team focused on Flux also joining. Do we need a room or would we just meet at the project booth?

**Reply by stefanprodan** [reply]
I think there is a space for project meetings close to the booth.

**Reply by LucaLanziani** [reply]
I'll be there and I'd love to connect

**Reply by vukg** [reply]
I would love to connect on KubeCon in Paris too. Maybe we could organize pop-up Flux users/supporters gathering / lunch.

**Reply by monadic** [reply]
https://kccnceu2024.sched.com/event/1akzm

**Comment by jainankit** [comment]
Hi, my name is Ankit Jain, founder of [Aviator](https://aviator.co) - a suite of developer collaboration tools. We are a small team of ex-Googlers looking to support The Flux project and the Flux users. We will allocate some resources to help out existing users, while finding ways to provide enterprise support and consulting services. If you are currently using or exploring the use of Flux, please reach out.

**Reply by monadic** [reply]
amazing

there are community meetings @mewzherder can explain!

**Comment by paulofrazao** [comment]
I can continue to update/patch Flux on Operatorhub.io.

**Comment by mestadler** [comment]
Paulo, that would be great 😃

**Comment by nagyv** [comment]
GitLab continues to support and build on Flux, and we are committed even more to playing an active role in the Flux community and we want to support FluxCD for enterprise customers. You can read the official announcement at https://about.gitlab.com/blog/2024/03/05/the-continued-support-of-fluxcd-at-gitlab/

Let me know if you have any questions or would like to discuss our decision further. You can reach me on the CNCF Slack as Viktor Nagy (Product Manager @ GitLab  🦊).

**Reply by monadic** [reply]
Viktor, please do let everyone know if anyone may be at Kubecon!

**Reply by nagyv** [reply]
Given other company events, GitLab will be present only at KubeCon US this year. There might be some Gitlab employees in KubeCon EU, but it's less organized than usual and I don't know specific names.

**Reply by abuango** [reply]
Hi @monadic, I am with the Developer Relations team at GitLan and I will be attending KubeCon EU in Paris.

**Reply by monadic** [reply]
@abuango https://kccnceu2024.sched.com/event/1akzm

**Reply by monadic** [reply]
please LMK if you can make it

**Comment by o6uoq** [comment]
Much like Crowded House and the weather, everywhere I go I will take Flux with me. Looking forward to a very bright future for this year and beyond for Flux and Friends 🤗

**Comment by monadic** [comment]
posted on linkedin https://www.linkedin.com/feed/update/urn:li:activity:7171213301555666945/

spread the word people

x

**Comment by dolobanko** [comment]
OpsWorks Group, that includes [Upplabs](https://www.linkedin.com/company/upplabs/), [OpsWorks](https://www.linkedin.com/company/opsworksco), and [RhodeCode](https://www.linkedin.com/company/rhodecode) is ready to support Flux customers and OSS development.

**Reply by monadic** [reply]
nice

**Comment by makkes** [comment]
It's really lovely to see so many people and companies chiming in here in support of the Flux project and with the intent to make it successful going forward. I'm speaking for myself here but I'm convinced these shouts encourage all Flux core maintainers a lot after the uncertainty caused by Weavework's folding yearlier in the year. Since then the Flux community already diversified a lot and some of us already found new jobs where we can keep working on Flux at least for a couple of hours a week.

What makes Flux and really any Open Source project successful are the people working each day on the software, the documentation, the community, the communication, helping our users, supporting them put Flux to work successfully. How can people keep working on Flux, though? That's only possible in two ways:

1. You work on it in your spare time after "coming home" from work. This is a very rare exception, though, because we want to have a life out of software, too, we have a family that we want to spend time with, we have friends, we want to enjoy a nice evening outside.
2. You get paid to work on Flux. This is the majority of people currently contributing to the project. Your company lets you work on Flux during your paid time, maybe for a couple of hours a week, maybe even half-time, sometimes (also rare) you can work on Flux full-time.

To all those who have chimed in here I want to say thank you and accompany that with a plea: If you want to see Flux thrive, please let your people work on it during their working hours. Make it very explicit how much time they will be able to spend contributing, make this time accountable and measurable so that they can very clearly say "today is my Flux day, no other meetings, please" or "every day between 2pm and 4pm I'm working on Flux" or anything like that. This is the way to make an Open Source project successful. This is the air that Flux breathes.

If you've been given time to work on Flux from your employer, then it's time to find something exciting to work on. We maintain [a roadmap](https://fluxcd.io/roadmap/) that gives you a first impression what we have planned in the coming months. Next, you may want to join the regular dev meetings and listen to what the maintainers are working on and to chime in with your desire to contribute. Find details on these meetings [here](https://github.com/fluxcd/community/?tab=readme-ov-file#meetings).

Flux lives from people actively working on it!

Cheers!
/max

**Comment by vukg** [comment]
Hello everyone 👋

My name is Vuk Gojnic from Deutsche Telekom. We are heavy users of Flux for our Cloud Native Telco platform since almost 5 years: https://github.com/telekom/das-schiff.

We are OSS end-user and we do not have customers consuming it directly. So we are contributing as end-user and have been working closely with Weaveworks in the past for enterprise support. As contingency planning we are analysing the alternatives, but overwhelming community committment to the project gives us the confidence and motivation to continue with Flux and support in the ways we can.

Having analyzed support options we have concluded that we are not right type of organization for maintainers of such important OSS project to work for. However we stay committed to support as OSS end-user.

-Vuk (Platform Lead Deutsche Telekom Technik)

**Comment by monadic** [comment]
Hi all, we have a room booked at Kubecon!   We have space on Wednesday, 20 March from 11:00 - 12:30. The room will hold 50 people with seating at square/communal tables.  There is standard AV in the room - screen, projector, 2 microphones.

The purpose is to assemble as many folks as possible from this group who are also at Kubecon.  The topic is Flux and Ecosystem, which means things like the Flux-iac project (tf-controller), GUIs, and more.  People should be able to help each other more after seeing the potential areas of activity.  (In other words the primary goal of the meet up is transparency and communication)

**Reply by muhlba91** [reply]
awesome idea! any information on what room number?

**Reply by puja108** [reply]
Great, I'm definitely in!

**Reply by monadic** [reply]
https://kccnceu2024.sched.com/event/1akzm

**Comment by TNGHK** [comment]
Greetings, 
I hope this helps a bit: We from TNG aka "The Nerd Group" have currently about 27 of our 832 people actively using Flux in their projects. 10 of those are on an advanced level regarding Flux. As we are a professional services organisation (for code, AI and cloud), we can also work on Flux-intensive projects if need be. OSS-wise, the biggest OSS project we are currently working on on Github is [Opossum](https://github.com/opossum-tool/OpossumUI/), other OSS stuff from us such as ArchUnit can be found [here](https://github.com/TNG). 
Cheers, Henrik

**Comment by vishnukumar356** [comment]
any update on weave-gitops ui ?, is it maintained in enterprise edition ?

**Reply by erikgb** [reply]
> any update on weave-gitops ui ?, is it maintained in enterprise edition ?

AFAIK, weave-gitops is currently unmaintained. And not (yet) forked. We are still hoping for someone to pick it up!

**Reply by stefanprodan** [reply]
weave-gitops UI is being ported to Headlamp https://github.com/headlamp-k8s/plugins/pull/75

**Reply by vishnukumar356** [reply]
Thanks. will it have feature to stop the reconciliation and resume it ?

**Comment by nagyv** [comment]
Just for the record, GitLab already offers support for FluxCD, built into the GitLab UI, including in the free edition.

Currently, GitLab shows the reconciliation status, and allows to trigger, suspend or resume a reconciliation from the UI.

Related documentation is available here: https://docs.gitlab.com/ee/ci/environments/kubernetes_dashboard.html

-------- Eredeti üzenet --------
2024. 09. 26. 12:39-kor, vishnukumar356 ezt írta:
